File: BaekJoon Online Judge/C++/Step10-Brute Force/2798.cpp
//
//  2798.cpp
//  Step10-Brute Force
//
//  Created by 김예빈 on 2019. 9. 5..
//  Copyright © 2019년 김예빈. All rights reserved.
//

#include <iostream>
using namespace std;

int main(int argc, const char * argv[]) {
    cin.tie(NULL);
    ios::sync_with_stdio(false);
    
    int n, max, sum, result = 0, i, j, k;
    
    cin >> n >> max;
    
    int card[n];
    
    for(i = 0; i < n; i++)
        cin >> card[i];
    
    for(i = n - 1; i >= 0; i--) {
        for(j = i - 1; j >= 0; j--) {
            for(k = j - 1; k >= 0; k--) {
                sum = card[i] + card[j] + card[k];
                
                if(sum <= max)
                    if(result < sum)
                        result = sum;
            }
        }
    }
    
    cout << result;
    
    return 0;
}

File: BaekJoon Online Judge/Swift/Step12-Brute Force/2798.swift
//
//  2798.swift
//  Step11-Brute Force
//
//  Created by Yebin Kim on 2019/11/26.
//  Copyright © 2019 김예빈. All rights reserved.
//

import Foundation

let input = readLine()!.split(separator: " ").map{ Int($0)! }
let card = readLine()!.split(separator: " ").map{ Int($0)! }.sorted(){ $0 > $1 }
let N = input[0]
let M = input[1]

var answer = 0

for i in 0..<(N - 2) {
    for j in (i + 1)..<(N - 1) {
        for k in (j + 1)..<N {
            let sum = card[i] + card[j] + card[k]
            
            if (sum <= M) && (answer < sum) {
                answer = sum
            }
        }
    }
}

print(answer)
